File: apps/mobile/test/check_in_window_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:imokay/domain/entities/check_in_window.dart';

void main() {
  test('formats the daily window', () {
    const window = CheckInWindow(
      id: 'w',
      userId: 'u',
      startHour: 8,
      startMinute: 0,
      endHour: 12,
      endMinute: 0,
    );
    expect(window.label, '08:00 – 12:00');
  });
}

File: apps/api/src/models/check-in-window.test.ts
import { describe, it } from "node:test";
import assert from "node:assert/strict";
import {
  createCheckInWindow,
  isWithinWindow,
} from "./check-in-window.js";

describe("createCheckInWindow", () => {
  it("rejects a window that does not move forward", () => {
    assert.throws(() =>
      createCheckInWindow({
        userId: "user-1",
        startHour: 12,
        startMinute: 0,
        endHour: 8,
        endMinute: 0,
      }),
    );
  });
});

describe("isWithinWindow", () => {
  const window = createCheckInWindow({
    userId: "user-1",
    startHour: 8,
    startMinute: 0,
    endHour: 12,
    endMinute: 0,
  });

  it("is true at 9:00", () => {
    const at = new Date(2026, 8, 8, 9, 0, 0);
    assert.equal(isWithinWindow(window, at), true);
  });

  it("is false at 12:00", () => {
    const at = new Date(2026, 8, 8, 12, 0, 0);
    assert.equal(isWithinWindow(window, at), false);
  });
});
